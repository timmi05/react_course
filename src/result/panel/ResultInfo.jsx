import React from 'react';

export const ResultInfo = (props) => {

    const { count, genre } = props;
    const massage = count ? `${ count } movie found` : genre ? `Film by ${ genre } genre` : '';

    return (
        <span className='result-info'>
        { massage }
        </span>
    );
}

export default ResultInfo