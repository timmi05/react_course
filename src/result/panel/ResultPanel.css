.result-panel{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 25px 30px 25px;
    font-size: 20px;
    margin: 0 40px;
}
