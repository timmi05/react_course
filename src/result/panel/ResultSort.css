.result-sort{
    display: flex;
}

.result-sort-button{
    font-size: 15px;
    padding: 10px 20px;
}

.result-sort-label{
    padding-right: 15px;
    display: flex;
    justify-content: center;
    flex-direction: column;
}