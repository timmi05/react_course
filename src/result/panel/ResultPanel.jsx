import React from 'react';
import {ResultInfo} from './ResultInfo';
import ResultSort from './ResultSort';
import './ResultPanel.css'

export const ResultPanel = props =>
    <div className='result-panel'>
        <ResultInfo
            {...props}
        />
        <ResultSort/>
    </div>;
