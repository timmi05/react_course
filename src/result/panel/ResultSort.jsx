import React from 'react';
import {Button} from '../../component/Button'
import './ResultSort.css'
import {connect} from 'react-redux'
import {selectIsProductSelected} from '../../store/selectors/selectIsProductSelected'
import {selectSortBy} from '../../store/selectors/selectSortBy'
import {openMovie} from '../../store/actions/sort'

const ResultSort = props => {
    const {onSortBy, sortBy, isProductSelected} = props;
    if (isProductSelected)
        return (
            <a/>
        );

    return (
        <span className='result-sort'>
        <label className='result-sort-label'>SORT BY </label>
            <div className='result-sort-buttons'>
                <Button
                    label='RELEASE DATE'
                    classes='result-sort-button result-sort-left-button'
                    onClick={() => onSortBy('release_date')}
                    disabled={sortBy === 'release_date'}
                />
                <Button
                    label='RATING'
                    classes='result-sort-button result-sort-right-button'
                    onClick={() => onSortBy('vote_average')}
                    disabled={sortBy === 'vote_average'}
                />
            </div>
        </span>
    );
};

const mapStateToProps = state => ({
    sortBy: selectSortBy(state),
    isProductSelected: selectIsProductSelected(state)
});

const mapDispatchToProps = dispatch => ({
    onSortBy: sortBy => dispatch(openMovie(sortBy))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ResultSort);
