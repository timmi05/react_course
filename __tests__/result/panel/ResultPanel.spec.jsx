import React from 'react';
import {shallow} from 'enzyme';
import {ResultPanel} from '../../../src/result/panel/ResultPanel';

describe('ResultPanel rendering', () => {
    it('renders correctly', () => {
        const component = shallow(<ResultPanel/>);
        expect(component).toMatchSnapshot();
    });
});
