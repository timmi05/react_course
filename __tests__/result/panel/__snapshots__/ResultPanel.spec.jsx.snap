// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPanel rendering renders correctly 1`] = `
<div
  className="result-panel"
>
  <ResultInfo />
  <Connect(ResultSort) />
</div>
`;
